scipy/optimize/_linprog.py

```diff
@@ -338,13 +344,30 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     This section describes the available solvers that can be selected by the
     'method' parameter.
 
+    :ref:`'highs-simplex' <optimize.linprog-highs-simplex>` and
+    :ref:`'highs-ipm' <optimize.linprog-highs-ipm>` are interfaces to the
+    HiGHS simplex and interior-point method solvers [13]_, respectively.
+    'highs' chooses between the two automatically. These are the fastest linear
+    programming solvers in SciPy, especially for large, sparse problems.
     :ref:`'interior-point' <optimize.linprog-interior-point>` is the default
-    as it is typically the fastest and most robust method.
+    as it was the fastest and most robust method before the recent
+    addition of the HiGHS solvers. This may change in a future release.
     :ref:`'revised simplex' <optimize.linprog-revised_simplex>` is more
-    accurate for the problems it solves.
+    accurate than interior-point for the problems it solves.
     :ref:`'simplex' <optimize.linprog-simplex>` is the legacy method and is
     included for backwards compatibility and educational purposes.
 
+    Method *highs-simplex* is a wrapper of the C++ high performance dual
+    revised simplex implementation (HSOL) [13]_, [14]_. Method *highs-ipm*
+    is a wrapper of a C++ implementation of an **i**\ nterior-\ **p**\ oint
+    **m**\ ethod [13]_; it features a crossover routine, so it is as accurate as
+    a simplex solver. These are the fastest methods available in SciPy,
+    especially for large, sparse problems; which of these two is faster is
+    problem-dependent. For new code, we recommend explicitly choosing one of
```

Comment:
There is no old code, so this is weird. If you want people to explicitly choose, please remove the auto-selection from this PR.

Reply:
> There is no old code, so this is weird.  

I don't understand. I make the distinction between "new" code under active development and "old" code, which is not actively developed. For "new" code involving `linprog`, a HiGHS method should be selected; I make no such recommendation for "old" code.

> I meant to explicitly choose `method='highs'`, `method='highs-simplex'`, `method='highs-ipm'`. Rewording to:
```
    Method *highs-simplex* is a wrapper of the C++ high performance dual
    revised simplex implementation (HSOL) [13]_, [14]_. Method *highs-ipm*
    is a wrapper of a C++ implementation of an **i**\ nterior-\ **p**\ oint
    **m**\ ethod [13]_; it features a crossover routine, so it is as accurate as
    a simplex solver. Method 'highs' chooses between the two automatically.
    For new code involving `linprog`, we recommend explicitly choosing one of
    these three method values.
```